geonode/harvesting/harvesters/geonode.py

```diff
@@ -38,14 +44,14 @@
 from ..utils import XML_PARSER
 from . import base
 
-from geonode.maps.models import Map
-from geonode.layers.models import Layer
-from geonode.base.models import ResourceBase
-from geonode.documents.models import Document
-
 logger = logging.getLogger(__name__)
 
 
+class GeoNodeLayerType(enum.Enum):
+    VECTOR = "vector"
```

Comment:
Notice that now GeoNode already exposes generic enumerations for the layer types here
https://github.com/geosolutions-it/geonode/blob/nexus_master/geonode/base/enumerations.py#L495

Moreover the `storeType` in the _new_ GeoNode model is changed, e.g. from the old `coverageStore` to the new one `raster`

Reply:
I'll take a look at it and adapt

Reply:
After checking the existing enumerations, I believe the layer type enumeration is not very convenient to use since it is using  a simple list of strings. This makes it hard to refer to a specific type of layer in the code.

I mean if I want to refer to the `raster` type I need to do

```python
LAYER_TYPES[0]
```

which makes it a bit harder to read the code. IMO this is better:

```python
GeoNodeLayerType.RASTER
```

Unless you feel very strongly about this, I would prefer to not change the current code.

I think I could do it the other way around: submit a PR to change the current layer enumeration from a list of strings to an `Enum` class. I would do this a bit later, against main GeoNode repo. How would that work for you @afabiani ?